src/Packagist/WebBundle/Controller/UserController.php

```diff
@@ -270,6 +274,92 @@ public function deleteFavoriteAction(User $user, Package $package)
         return new Response('{"status": "success"}', 204);
     }
 
+    /**
+     * @Template()
+     * @Route("/users/{name}/2fa/", name="user_2fa_configure", methods={"GET"})
+     * @ParamConverter("user", options={"mapping": {"name": "username"}})
+     */
+    public function configureTwoFactorAuthAction(User $user)
+    {
+        if (!($this->isGranted('ROLE_DISABLE_2FA') || $user->getId() === $this->getUser()->getId())) {
+            throw new AccessDeniedException('You cannot change this user\'s two-factor authentication settings');
+        }
+
+        if ($backupCode = $this->get('session')->get('backup_code')) {
```

Comment:
I briefly considered using a flash message for this, but I wanted special formatting and therefore opted to just stash the code in normal session storage and manually remove it once we read it.